### PR TITLE
feat: next 3 lessons downloads - LESQ-419

### PR DESCRIPTION
**Pull request by JBR90**
## Description

I want to update the Downloads page to provide [up to] the next three lessons in the unit.
This PR just adds the data.

- Adds getNextLessonsInUnit()
- Uses lesson listing query on downloads page to fetch unit (api and api-2023)
- used getNextLessonsInUnit() to provide lessonTitle and slug for up to next three units.
- Adds schema / tests 

## How to test

This new functionality should be tested by QA when connected up to front end. Downloads page should work as expected

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { LessonListSchema } from "./lessonDownloads.schema";
+
+const getNextLessonsInUnit = (unit: LessonListSchema, lessonSlug: string) => {
+  if (!unit || !lessonSlug) {
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts:4]: can this happen?

File: src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+  const lessonInUnit = unit.find((lesson) => lesson.lessonSlug === lessonSlug);
+  const lessonPosition = lessonInUnit?.orderInUnit;
+  const nextLessons =
+    lessonPosition &&
+    unit
+      .filter((lesson) =>
+        lesson.orderInUnit ? lesson.orderInUnit > lessonPosition : [],
+      )
+      .splice(0, 3);
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts:15]: are the lessons always going to be sorted by orderInUnit?

File: src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.test.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    expect(nextLessons[0]?.lessonSlug).toEqual("subtract-three-surds");
+    expect(nextLessons[1]?.lessonSlug).toEqual("subtract-four-surds");
+  });
+
+  test("returns the next lesson when the lesson is the 2nd from the last in the unit", async () => {
+    const nextLessons = getNextLessonsInUnit(lessons, "subtract-three-surds");
+    expect(nextLessons.length).toEqual(1);
+    expect(nextLessons[0]?.lessonSlug).toEqual("subtract-four-surds");
+  });
+  test("returns the next lesson when the lesson is the 1nd and there are only 2 lessons in the unit", async () => {
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.test.ts:41]: slight typo

**Review by k-huggs — APPROVED** [review]
Small typo but other LGTM ✅

**Review by njblore — CHANGES_REQUESTED** [review]
couple of qs!

File: src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts (hunk 2 above)
Comment (JBR90) [reply src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts:15]: In the old API we order in the GQL query in the new API it's ordered in the MV

File: src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts (hunk 1 above)
Comment (JBR90) [reply src/node-lib/curriculum-api-2023/queries/lessonDownloads/getNextLessonsInUnit.ts]: No have removed

**Review by njblore — APPROVED** [review]

**Pull request merged**